### PR TITLE
Generic checkout - Delivery fields

### PR DESCRIPTION
**Pull request by jamesgorrie**
## What are you doing in this PR?
Adds delivery address fields to the generic checkout.

Currently this just uses those fields to fill the `billingAddress` prop. We'll get onto that next.

Mostly it's using components and functionality that already exists.

## Next steps

- Ensure country and state selector follow the logic on current checkouts
- Add billing address and ensure the data sent to the server matches current checkouts

[**Trello Card**](https://trello.com/c/tUUScZqY/674-add-delivery-and-billing-address-fields-to-checkout)

## Screenshots
https://github.com/guardian/support-frontend/assets/31692/fed5bc74-ccd8-4cc2-ae48-7fd697969a7d

URL
https://support.thegulocal.com/uk/checkout?product=HomeDelivery&ratePlan=Sixday

Part of #5722

### CODE REVIEW

File: support-frontend/assets/components/subscriptionCheckouts/address/postcodeFinder.tsx [hunk 1]
```diff
@@ -114,7 +114,10 @@ function AddressSelector({ results, onAddressSelected }: AddressSelectorProps) {
 			<Option value={''}>Select an address</Option>
 			<>
 				{results.map((result, index) => (
-					<Option key={result.lineOne} value={index}>
+					<Option
+						key={`${result.lineOne ?? ''}/${result.lineTwo ?? ''}`}
```
Comment (jamesgorrie) [inline support-frontend/assets/components/subscriptionCheckouts/address/postcodeFinder.tsx:118]: We needed to be more specific here as I was getting the "Two children with the same key" error.

File: support-frontend/assets/pages/[countryGroupId]/checkout.tsx [hunk 2]
```diff
@@ -168,10 +181,18 @@ function describeProduct(product: string, ratePlan: string) {
 		}
 	}
 
+	if (product === 'NationalDelivery') {
```
Comment (paul-daniel-dempsey) [inline support-frontend/assets/pages/[countryGroupId]/checkout.tsx:184]: Is `NationalDelivery` the same as `HomeDelivery`  - ie could they be combined?

File: support-frontend/assets/pages/[countryGroupId]/checkout.tsx [hunk 3]
```diff
@@ -439,8 +494,66 @@ export function Checkout() {
...
+												setCountry={() => {
+													// no-op
+												}}
+												setPostcodeForFinder={() => {
+													// no-op
+												}}
+												setPostcodeErrorForFinder={() => {
+													// no-op
+												}}
+												onFindAddress={(postcode) => {
```
Comment (paul-daniel-dempsey) [inline support-frontend/assets/pages/[countryGroupId]/checkout.tsx:543]: Could'nt get this to 'Find Address'? Does it give you more than empty results?

**Review by paul-daniel-dempsey — COMMENTED** [review]
Does the PostCode 'FindAddress' work in development now? I tried to run but had no luck with results for something like `N1 9GU`? Delivery address fields all look good.

File: support-frontend/assets/pages/[countryGroupId]/checkout.tsx (hunk 2 above)
Comment (jamesgorrie) [reply support-frontend/assets/pages/[countryGroupId]/checkout.tsx:184]: I am trying keep things decoupled at the moment making things easy to change going forward.

File: support-frontend/assets/pages/[countryGroupId]/checkout.tsx (hunk 3 above)
Comment (jamesgorrie) [reply support-frontend/assets/pages/[countryGroupId]/checkout.tsx:543]: This doesn't locally or in `CODE`.

I stubbed it out for testing, but we should look at getting this running in future.

**Review by paul-daniel-dempsey — APPROVED** [review]
Looks Great,

**Pull request merged**